src/server/services/wms/qgswmsparameters.cpp

```diff
@@ -1492,6 +1498,22 @@ namespace QgsWms
     return params;
   }
 
+  QList<QgsWmsParametersExternalLayer> QgsWmsParameters::externalLayersParameters() const
+  {
+    QList<QgsWmsParametersExternalLayer> externalLayers;
+
+    const QStringList layers = allLayersNickname();
+    for ( const QString &layer : allLayersNickname() )
+    {
+      if ( ! isExternalLayer( layer ) )
```

Comment:
If you feel like playing with c++11 std algorithms this is a great candidate for a std::remove_if type filter :wink: